### PR TITLE
Generating documentation for TorchData

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #193

This PR contain changes that will generate documentation for the library. Other improvements will be needed. The follow-ups are being tracked here #195. Those issues can be addressed in follow up PRs.

Differential Revision: [D33986028](https://our.internmc.facebook.com/intern/diff/D33986028)

### CODE REVIEW

**Review by ejguan — APPROVED** [review]
Thanks for adding this.
Could we have a visualized output on pages before landing? Or we have to build it on our local machine?

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          make html 
+          cd ..
+      - name: Get output time
+        run: echo "The time was ${{ steps.build.outputs.time }}"
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@releases/v3
+        with:
+          ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BRANCH: gh-pages # The branch the action should deploy to.
+          FOLDER: docs/build/html # The folder the action should deploy.
```
[the remark this answers is not in the document]
Comment (NivekT) [reply .github/workflows/docs.yml:38]: I do not believe so, we will have to turn on "GitHub Pages Enabled" in our internal setting (which has not been set up yet). Before then, we will have to build the doc locally to see it.

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33986028).

File: .github/workflows/docs.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ejguan) [reply .github/workflows/docs.yml:38]: If so, I will test it tomorrow.

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33986028).

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+def setup(app):
+
+    # Overwrite class name to allow aliasing in documentation generation
+    import torchdata.datapipes.iter as iter
+    import torchdata.datapipes.map as map
+
+    for mod in (iter, map):
+        for name, obj in mod.__dict__.items():
+            if isinstance(obj, type):
+                obj.__name__ = name
```
Comment (wenleix) [inline docs/source/conf.py:95]: noob question: how does this allow aliasing in documentation generation? ^_^

Comment (wenleix) [reply docs/source/conf.py:95]: @NivekT  I see ! Thanks for the explanation!

**Pull request closed without merging**